src/lib/tls/tls12/msg_server_kex.cpp

```diff
@@ -60,20 +60,28 @@ Server_Key_Exchange::Server_Key_Exchange(Handshake_IO& io,
 
       if(dh_groups.empty())
          {
-         shared_group = policy.default_dh_group();
+         m_shared_group = policy.default_dh_group();
          }
       else
          {
-         shared_group = policy.choose_key_exchange_group(dh_groups, {});
+         m_shared_group = policy.choose_key_exchange_group(dh_groups, {});
          }
 
-      if(shared_group == Group_Params::NONE)
+      if(m_shared_group.value() == Group_Params::NONE)
          throw TLS_Exception(Alert::HandshakeFailure,
                "Could not agree on a DH group with the client");
 
-      BOTAN_ASSERT(group_param_is_dh(shared_group), "DH groups for the DH ciphersuites god");
-
-      m_kex_key = state.callbacks().tls_generate_ephemeral_key(shared_group, rng);
+      BOTAN_ASSERT(group_param_is_dh(m_shared_group.value()), "DH groups for the DH ciphersuites good");
```

Comment:
I'm sure you thought "god" was a typo, actually was a reference to https://knowyourmeme.com/memes/blood-for-the-blood-god

We should change this to something reasonable eg "DH ciphersuite is using a finite field group"

Reply:
Yikes! 🤣 made my day.